### ISSUE TITLE
Filter functions to start in the emulator

### ISSUE THREAD

**Issue by SamyPesse** [firebase/firebase-tools#2340]
Right now there is no way to do only start the emulator with a subset of functions enabled instead of all. It requires a custom build or filter the exports ourself.

It'd be great if the `firebase emulators:exec` and `firebase emulators:start` would accept an argument like `--functions a,b` or maybe the same syntax as `firebase deploy`: `--only functions:a,functions:b` (this is what I was expecting at first)

**Comment by samtstern** [comment]
@SamyPesse this is a reasonable feature request and should be feasible to implement.  Can you explain the reason that you don't want to emulate all functions though?  Curious if there's an underyling emulator improvement we could make.

**Comment by SamyPesse** [comment]
I'm basically running a script `generate-tests-data` to prepare the dataset for unit testing:

```bash
export CF_IS_PREPARING_DATASET="true"

firebase emulators:exec --only firestore,functions \
    "yarn node write-to-emulator.ts && \
    sleep 60 && \
    curl http://localhost:8080/emulator/v1/projects/$PROJECT_ID:ruleCoverage.html > tests/coverages/firestore.html && \
    firebase emulators:export ./dump/tests --force"
```

To explain this code:
- We start the `functions` emulator because some documents are "computed" in cloud functions
- We use `CF_IS_PREPARING_DATASET` to have different exports and not run all functions, because only a few are used to compute those data, and most other functions depend on external services
- We use `sleep 60` to wait for the end of the CF executions 🙃
- We use the `curl` to access the coverage after executing the command

The improvements I'd like from Firebase are:

- Specify functions to run in `firebase emulators:exec` to avoid the use of an env `CF_IS_PREPARING_DATASET` to return a different set of functions
- A command `firebase emulators:functions:wait` to wait for all cloud functions executions to be finished (similar to https://github.com/firebase/firebase-tools/issues/1813)

**Comment by samtstern** [comment]
@SamyPesse I just sent a PR to fix #1813 which will be included in the next release.  We'll now wait up to 60s for any functions to finish when the emulators are stopped.

**Comment by samtstern** [comment]
@SamyPesse makes sense I actually end up in this situation a lot.  Would it be useful for you if the emulator always wrote the rules coverage report to a file on exiting?  Or do you prefer to have more control over when/where that happens?

**Comment by SamyPesse** [comment]
@samtstern yes it'll fix it 👍
